### ISSUE TITLE
Getting this plugin upstream

### ISSUE THREAD

**Issue by objectified** [lorin/ansible-docker-connection#5]
Hi @lorin ,

Thanks for writing this excellent plugin. I took your source and tested it against the current v2 devel upstream branch for Ansible, and a few things broke. I fixed those things, and now I'd like to get it merged into Ansible. I'm not sure if that will succeed, but for now I've created an Ansible feature branch with the v2 compatible plugin in it (https://github.com/objectified/ansible/tree/feature-docker-connection/lib/ansible/plugins/connections). I've taken the liberty of adding my name to the copyright header of the source file.

Since you've written most of it - I'd like to ask your approval of this. Are you okay with me going forward and try to get this into Ansible itself? I'd be fine with you doing it instead (taking my version), but either way I'd like it there  :)

Please let me know.

**Comment by lorin** [comment]
Sure, please feel free to submit the fixed version upstream.

Note that once https://github.com/docker/docker/pull/13171 lands, Docker will have native support for copying files in, and the put_file method can then be modified to use a proper Docker call rather than the existing workaround.

**Comment by objectified** [comment]
Excellent, thanks.

I'm aware of the fix in Docker with regards to `docker cp`, that would be very nice to see implemented. I suppose that in the meantime the cat workaround should do, especially for files that aren't all too large.

**Comment by objectified** [comment]
Just dropping by to notify that this plugin got merged after a number of modifications. As for your last comment - the plugin now uses "native" docker cp when the Docker version is higher than 1.8.0; some other work involved getting the plugin ready for Ansible v2. 

https://github.com/ansible/ansible/pull/11650

Thanks again for the initial work on the plugin!
